app/controllers/registration_commands_controller.rb

```diff
@@ -44,6 +44,11 @@ def registration_params
     end
   end
 
+  def host_groups_json
+    Hostgroup.authorized(:view_hostgroups)
+             .map { |hg| hg.as_json(methods: :inherited_operatingsystem_id) }
```

Comment:
The only question I have with this is how queries it performs. Is it just 1 or 1+n?

Reply:
It seems okay, `bullet` is not saying anything and from logs only two selects have been done:
```
Hostgroup Load (0.7ms)  SELECT "hostgroups".* FROM "hostgroups" WHERE (hostgroups.id IN (1)) ORDER BY hostgroups.title
app/controllers/registration_commands_controller.rb:49:in `map'
Operatingsystem Load (0.5ms)  SELECT "operatingsystems".* FROM "operatingsystems" WHERE "operatingsystems"."id" = $1 ORDER BY "operatingsystems"."title" ASC LIMIT $2  [["id", 1], ["LIMIT", 1]]
app/models/concerns/nested_ancestry_common.rb:68:in `block (2 levels) in nested_attribute_for'
```

Reply:
And if we have some issues with efficiency, we should fix it in https://github.com/theforeman/foreman/blob/26541260875c837832f033e6897ab9673c7b6be2/app/models/concerns/nested_ancestry_common.rb#L41, if the inherited field is required, it should be ok to include it.